src/Akeneo/Bundle/ElasticsearchBundle/Cursor/FromSizeCursor.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Akeneo\Bundle\ElasticsearchBundle\Cursor;
+
+use Akeneo\Bundle\ElasticsearchBundle\Client;
+use Akeneo\Component\StorageUtils\Cursor\CursorInterface;
+use Akeneo\Component\StorageUtils\Repository\CursorableRepositoryInterface;
+
+/**
+ * Bounded cursor to iterate over items where a start and a limit are defined.
+ * Internally, this is implemented with the from/size pagination.
+ * {@see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/search-request-from-size.html}
+ *
+ * @author    Julien Janvier <jjanvier@akeneo.com>
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FromSizeCursor extends AbstractCursor implements CursorInterface
+{
+    /** @var int */
+    protected $initialFrom;
+
+    /** @var int */
+    protected $from;
+
+    /** @var int */
+    protected $limit;
+
+    /** @var int */
+    protected $to;
+
+    /** @var int */
+    protected $fetchedItemsCount;
+
+    /**
+     * @param Client                        $esClient
+     * @param CursorableRepositoryInterface $repository
+     * @param array                         $esQuery
+     * @param string                        $indexType
+     * @param int                           $pageSize
+     * @param int                           $limit
+     * @param int                           $from
+     */
+    public function __construct(
+        Client $esClient,
+        CursorableRepositoryInterface $repository,
+        array $esQuery,
+        $indexType,
+        $pageSize,
+        $limit,
+        $from = 0
+    ) {
+        $this->esClient = $esClient;
+        $this->repository = $repository;
+        $this->esQuery = $esQuery;
+        $this->indexType = $indexType;
+        $this->pageSize = $pageSize;
+        $this->limit = $limit;
+        $this->from = $from;
+        $this->initialFrom = $from;
+        $this->to = $this->from + $this->limit;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function next()
+    {
+        if (false === next($this->items)) {
+            $this->from += count($this->items);
+            $this->items = $this->getNextItems($this->esQuery);
+            reset($this->items);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * {@see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/search-request-from-size.html}
+     */
+    protected function getNextIdentifiers(array $esQuery)
+    {
+        $size = ($this->to - $this->from) > $this->pageSize ? $this->pageSize : ($this->to - $this->from);
+        $esQuery['size'] = $size;
+
+        if (0 === $esQuery['size']) {
+            return [];
+        }
+
+        $sort = ['_uid' => 'asc'];
+
+        if (isset($esQuery['sort'])) {
+            $sort = array_merge($esQuery['sort'], $sort);
```

Comment:
we want to be sure results are always ordered at least by _uid

cf. my previous comment